lib/ansible/modules/tacp/tacp_info.py

```diff
@@ -0,0 +1,221 @@
+                         "tag",
+                         "template",
+                         "user",
+                         "vlan",
+                         "vnet"]}
+    }
+
+    result = dict(
+        changed=False,
+        args=[]
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    if module.check_mode:
+        module.exit_json(**result)
+
+    result['args'] = module.params
+
+    # Define configuration
+    configuration = tacp.Configuration()
+    configuration.host = "https://manage.cp.lenovo.com"
+    configuration.api_key_prefix['Authorization'] = 'Bearer'
+    configuration.api_key['Authorization'] = module.params['api_key']
+    api_client = tacp.ApiClient(configuration)
+
+    resource_dict = {"application": tacp_utils.ApplicationResource,
+                     "application_group": tacp_utils.ApplicationGroupResource,
+                     "category": tacp_utils.CategoryResource,
+                     "datacenter": tacp_utils.DatacenterResource,
+                     "firewall_profile": tacp_utils.FirewallProfileResource,
+                     "instance": tacp_utils.ApplicationResource,
+                     "marketplace_template": tacp_utils.MarketplaceTemplateResource,  # noqa
+                     "migration_zone": tacp_utils.MigrationZoneResource,
+                     "site": tacp_utils.SiteResource,
+                     "storage_pool": tacp_utils.StoragePoolResource,
+                     "tag": tacp_utils.TagResource,
+                     "template": tacp_utils.TemplateResource,
+                     "user": tacp_utils.UserResource,
+                     "vlan": tacp_utils.VlanResource,
+                     "vnet": tacp_utils.VnetResource
+                     }
+
+    resource = resource_dict[module.params['resource']](api_client)
+
+    items = [item.to_dict() for item in resource.filter()]
+
+    if module.params['resource'] == 'migration_zone':
+        application_resource = tacp_utils.ApplicationResource(api_client)
+        category_resource = tacp_utils.CategoryResource(api_client)
+        datacenter_resource = tacp_utils.DatacenterResource(api_client)
+        for item in items:
+            if 'applications' in item:
+                uuids = [application['uuid'] for application in item['applications']]  # noqa
+                application_list = [application.to_dict() for application in
+                                    application_resource.filter(uuid=('=in=', *uuids))]  # noqa
+                for application in item['applications']:
+                    for app in application_list:
+                        if application['uuid'] == app['uuid']:
+                            application['name'] = app['name']
+                            break
+
+            if 'allocations' in item:
+                if 'categories' in item['allocations']:
+                    uuids = [category['uuid'] for category in item['categories']]  # noqa
```

Comment:
You are repeating this same series of steps over and over. You should make a function and make it generic.